### PR TITLE
s/env/envVars

### LINKED ISSUE #2893: [ci.jenkins.io] Container agents in a degraded state
### Service(s)

ci.jenkins.io

### Summary

I've now run 3 core PR builds in a row against a Linux JDK 8 Kubernetes container (label: `maven`) and all three have failed with:

```
Cannot contact jnlp-maven-8-45549: hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@3ba7fb1d:JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651": Remote call on JNLP4-connect connection from ec2-3-143-84-83.us-east-2.compute.amazonaws.com/3.143.84.83:64651 failed. The channel is closing down or has closed down`
```

- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/2/console
- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/3/console
- https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6499/4/console

https://status.jenkins.io/ shows no issues.

Clearly the cluster is in a severely degraded state and monitoring is insufficient.

### Reproduction steps

Try to run a core PR build. Observe that the build fails within a minute rather than compiling and testing the code.

### PR DESCRIPTION
**Pull request by basil**
Amends #2128. Fixes https://github.com/jenkins-infra/helpdesk/issues/2893#issuecomment-1104390064. CC @dduportal

### CODE REVIEW

File: dist/profile/templates/buildmaster/casc/clouds.yaml.erb [hunk 1]
```diff
@@ -160,7 +160,7 @@ jenkins:
         - containers:
           - alwaysPullImage: false
             image: "<%= @agent_images["container_images"][agent["name"]] %>"
-            env:
+            envVars:
             - name: "JENKINS_JAVA_OPTS"
               value: "-XX:+PrintCommandLineFlags"
```
Comment (dduportal) [inline dist/profile/templates/buildmaster/casc/clouds.yaml.erb]:
```suggestion
            envVars:
              - envVar:
                   key: "JENKINS_JAVA_OPTS"
                   value: "-XX:+PrintCommandLineFlags"
```

As per https://github.com/jenkinsci/configuration-as-code-plugin/issues/204#issuecomment-389623788 

(I did the same as you and had another stacktrace. Tried with the local vagrant VM used to test the puppet template, went through the Jenkins UI, set up manually the env vars, and checked the jcasc export).

**Review by dduportal — APPROVED** [review]
:thanks!

**Comment by basil** [comment]
Sorry for the trouble. I thought this was just standard Kubernetes YAML syntax, but now I see that the Jenkins Kubernetes plugin has its own syntax. Next time I'll test that syntax out in the Jenkins UI and get an export from JCasC.

**Comment by dduportal** [comment]
No worries @basil . The "checking" process on this repository is not really easy and it does not check for validity of this (it's an eternal problem with JCasc alas). Could be greatly improved by switching from vagrant VM to full container.

That one was tricky, and there are missing elements. It happens :)

**Pull request merged**